src/libs/collect.c

```diff
@@ -1034,153 +1111,158 @@ static void list_view(dt_lib_collect_rule_t *dr)
-                 "SELECT DISTINCT value, 1 FROM main.meta_data WHERE key = %d ORDER BY value",
-                 DT_METADATA_XMP_DC_PUBLISHER);
+                 "SELECT value, 1, COUNT(*) AS count FROM main.images JOIN "
+                         "(SELECT id AS meta_data_id, value FROM main.meta_data WHERE key = %d) ON id = meta_data_id "
+                         "WHERE %s GROUP BY value ORDER BY value",
+                 DT_METADATA_XMP_DC_PUBLISHER, where_ext);
         break;
       case DT_COLLECTION_PROP_RIGHTS: // rights
         snprintf(query, sizeof(query),
-                 "SELECT DISTINCT value, 1 FROM main.meta_data WHERE key = %d ORDER BY value ",
-                 DT_METADATA_XMP_DC_RIGHTS);
+                 "SELECT value, 1, COUNT(*) AS count FROM main.images JOIN "
+                         "(SELECT id AS meta_data_id, value FROM main.meta_data WHERE key = %d) ON id = meta_data_id "
+                         "WHERE %s GROUP BY value ORDER BY value",
+                 DT_METADATA_XMP_DC_RIGHTS, where_ext);
         break;
       case DT_COLLECTION_PROP_LENS: // lens
-        g_strlcpy(query, "SELECT DISTINCT lens, 1 FROM main.images ORDER BY lens", sizeof(query));
+        g_snprintf(query, sizeof(query), "SELECT lens, 1, COUNT(*) AS count "
+                "FROM main.images WHERE %s GROUP BY lens ORDER BY lens", where_ext);
         break;
 
       case DT_COLLECTION_PROP_FOCAL_LENGTH: // focal length
-        g_strlcpy(query, "SELECT DISTINCT CAST(focal_length AS INTEGER) AS focal_length, 1 "
-                         "FROM main.images ORDER BY focal_length",
-                  sizeof(query));
+        g_snprintf(query, sizeof(query), "SELECT CAST(focal_length AS INTEGER) AS focal_length, 1, COUNT(*) AS count "
+                         "FROM main.images WHERE %s GROUP BY focal_length ORDER BY focal_length",
+                   where_ext);
         break;
 
       case DT_COLLECTION_PROP_ISO: // iso
-        g_strlcpy(query, "SELECT DISTINCT CAST(iso AS INTEGER) AS iso, 1 FROM main.images ORDER BY iso",
-                  sizeof(query));
+        g_snprintf(query, sizeof(query), "SELECT CAST(iso AS INTEGER) AS iso, 1, COUNT(*) AS count "
+                           "FROM main.images WHERE %s GROUP BY iso ORDER BY iso",
+                   where_ext);
         break;
 
       case DT_COLLECTION_PROP_APERTURE: // aperture
-        g_strlcpy(query, "SELECT DISTINCT ROUND(aperture,1) AS aperture, 1 FROM main.images ORDER BY aperture",
-                  sizeof(query));
+        g_snprintf(query, sizeof(query), "SELECT ROUND(aperture,1) AS aperture, 1, COUNT(*) AS count "
+                           "FROM main.images WHERE %s GROUP BY aperture ORDER BY aperture",
+                   where_ext);
         break;
 
       case DT_COLLECTION_PROP_FILENAME: // filename
-        g_strlcpy(query, "SELECT DISTINCT filename, 1 FROM main.images ORDER BY filename", sizeof(query));
-        break;
-
-      case DT_COLLECTION_PROP_DAY:
```

Comment:
`DT_COLLECTION_PROP_{DAY,TIME}` are no longer handled. Is the default case working for those?